src/Microsoft.AspNetCore.Razor.Language/LargeTextSourceDocument.cs

```diff
@@ -17,8 +18,9 @@ internal class LargeTextRazorSourceDocument : RazorSourceDocument
         private readonly RazorSourceLineCollection _lines;
 
         private readonly int _length;
+        private byte[] _checksum;
 
-        public LargeTextRazorSourceDocument(StreamReader reader, int chunkMaxLength, Encoding encoding, string fileName)
+        public LargeTextSourceDocument(StreamReader reader, int chunkMaxLength, Encoding encoding, string fileName)
```

Comment:
Hm? We computed the stream bytes beforehand which means the stream reader can do whatever it wants. The `StreamSourceDocument` is the one that generates the proper checksum because its the one that needs the BOM bytes, not the `LargeTextSourceDocument`.

I may be misunderstanding your concern.